ISISComputingGroup/ibex_gui
# Ticket1253 rethink bumpstrip

### Description of work

Overhauled the spangle banner to display instrument specific properties defined in the blockserver in a more generic way. Previously the banner just had a display for the bumpstrip status hardcoded in. Now it can support any Boolean property, defined in the blockserver by having a name and pv, as well as true, false and unknown states with associated text message and -colour for display in the banner.

### To test

https://github.com/ISISComputingGroup/IBEX/issues/1253

The blockserver does not yet provide a sensible way to define banner items. Meanwhile you can test this PR with a hardcoded banner item in the blockserver. Check out the relevant branch (https://github.com/ISISComputingGroup/EPICS-inst_servers/pull/65) and add the following lines to the init method:

        # Create banner object
        bool_str = BoolStr("test property", "DAE:TITLE:DISPLAY")
        t_state = {"colour": "GREEN", "message": "displaying"}
        f_state = {"colour": "RED", "message": "not displaying"}
        u_state = {"colour": "RED", "message": "unknown"}
        bool_str.set_true_state(t_state)
        bool_str.set_false_state(f_state)
        bool_str.set_unknown_state(u_state)
        self.banner = Banner()
        self.banner.add_item(bool_str)

This creates a banner item pointing at the display title PV which can be toggled in the DAE view. You can play around with this, change parameters or add additional items to the banner etc.

### Acceptance criteria

*List the acceptance criteria for the PR*

1. The banner contains the correct items as defined in the blockserver
1. The banner items display the correct state information
1. The banner contains the appropriate items after switching / restarting an instrument.

#### Code Review


### Functional Tests
